ts/Extensions/Annotations/EventEmitter.ts

```diff
@@ -305,7 +305,16 @@ abstract class EventEmitter {
         }
 
         const emitter = this,
-            pointer = emitter.chart.pointer;
+            pointer = emitter.chart.pointer,
+            // Using experimental property on event object to check if event was
+            // created by touch on screen on hybrid device (#18122)
+            firesTouchEvents = pick(
+                (
+                    (e as any).sourceCapabilities &&
+                    (e as any).sourceCapabilities.firesTouchEvents
```

Comment:
Optional chaining won't work, because that property doesn't exist in Firefox and it shows errors in the console if used (?), first worth checking if `sourceCapabilities`.

`||` sounds like a good plan.

Reply:
After discussion and testing: optional chaining works and added to code.